File: di_java/src/main/java/com/kmmania/hr/DAOImplHR.java
package com.kmmania.hr;

import com.kmmania.api.DAO;

/**
 * Implementation of DAO interface for HR (heart rate) data.
 *
 * @author Dac Linh Giao NGUYEN
 * @version 0.1
 */
public class DAOImplHR implements DAO {
    /**
     * Retrieves the maximum heart rate (HR) from the HR data source.
     *
     * @return the maximum HR as a double.
     */
    @Override
    public double getData() {
        System.out.println("Data source provides a max heart rate 185bpm");
        return 185;
    }
}

File: di_kotlin/src/main/kotlin/hr/DAOImplHR.kt
/*
 * Dependency injection project.
 * Dac Linh Giao NGUYEN
 * License MIT
 *
 * Dependency injection project in Kotlin.
 * DAOImplHR.kt
 */
package com.kmmania.hr

import com.kmmania.api.DAO

/**
 * Implementation of the [DAO] interface for HR (heart rate) data.
 *
 * This class provides functionality to retrieve the maximum heart rate from the HR data source.
 *
 * @constructor Creates a [DAOImplHR] instance.
 * @author Dac Linh Giao NGUYEN
 * @since 0.1
 * @see DAO
 */
class DAOImplHR : DAO {
    /**
     * Retrieves the maximum heart rate (HR) from the HR data source.
     *
     * @return the maximum HR as a Double.
     */
    override fun getData(): Double {
        println("Data source provides a max heart rate 185bpm")
        return 185.0
    }
}